cpp/src/arrow/compute/kernels/codegen_internal.h

```diff
@@ -270,6 +275,24 @@ struct OutputArrayWriter<Type, enable_if_c_number_or_decimal<Type>> {
   }
 };
 
+template <typename Type>
+struct OutputArrayWriter<Type, enable_if_struct<Type>> {
```

Comment:
Hmm, I'm frankly not sure that any of this works. A Struct array isn't an array of Struct scalars, it's a nested array with one child for each struct field. For instance, there isn't a buffer 1 in a struct array at all, only a validity bitmap. This might still be made to work but it would have a lot of indirections around wrapping/unwrapping scalars. It might be better to specialize DivMod or consider a separate pattern for struct-returning arithmetic functions.

Reply:
Thanks! I reached the same conclusion. I will try to provide support for struct-returning arithmetic functions.